src/CMakeLists.txt

```diff
@@ -46,7 +46,23 @@ file (GLOB_RECURSE CPP_FILES *.cpp)
 file (GLOB_RECURSE H_FILES *.h)
 set (SOURCE_FILES ${CPP_FILES} ${H_FILES})
 
-add_executable(osp-magnum ${CPP_FILES})
+if(MSVC)
+	# For some reason, source_group() requires all files to be added to the
+	# executable. I did this conditionally in case it messes up other platforms
+	set(SHADERS_DIR "../bin/OSPData/adera/Shaders")
+  	file (GLOB_RECURSE SHADER_FILES
+  		"${SHADERS_DIR}/*.vert"
+  		"${SHADERS_DIR}/*.frag"
+  		"${SHADERS_DIR}/*.comp"
+  		)
+  	list(APPEND SOURCE_FILES ${SHADER_FILES})
+	add_executable(osp-magnum ${SOURCE_FILES})
```

Comment:
Better might be
```cmake
add_executable(osp-magnum)
file (GLOB_RECURSE H_FILES CONFIGURE_DEPENDS *.h)
file (GLOB_RECURSE CPP_FILES CONFIGURE_DEPENDS *.cpp)
target_sources(osp-magnum "${CPP_FILES}" "${H_FILES}")

# For some reason, source_group() requires all files to be added to the executable
	set(SHADERS_DIR "../bin/OSPData/adera/Shaders")
  	file (GLOB_RECURSE SHADER_FILES CONFIGURE_DEPENDS
  		"${SHADERS_DIR}/*.vert"
  		"${SHADERS_DIR}/*.frag"
  		"${SHADERS_DIR}/*.comp"
  		)
target_sources(osp-magnum "${SHADER_FILES}")
```

Reply:
And I wouldn't make it conditional on the platform. Unless someone adds a toolchain file to cmake that teaches cmake how to compile .vert, .frag and .comp files, cmake will ignore them.